docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+The goals for the features is to automate the following steps when building a model to achieve the above business goals:
+
+**Foundational features:** 
+
+- Provide an end-to-end **ML.NET CLI** for developers (i.e. *"mlnet new"*) to generate either the final trained model and the pipeline's C#/ML.NET implementation code in a similar fashion to the [.NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/?tabs=netcore2x). The CLI is also a foundation upon which higher-level tools, such as Integrated Development Environments (IDEs) can rest.
+- Automatic selection of best columns (input variables) to be used by the model
+- Automatic featurization and feature engineering 
+- Automatic learner sweeping and selection
+- Automatic hyperparameter sweeping and selection
+  
+
+# Solution
+
+The solution that can target any kind of developer and fits especially well when generating code is a CLI (Command-Line Interface). 
+
+The main support reasons are:
+
+- A CLI can be the foundation where other higher level tooling (UI, IDEs) can also rest on.
+- A CLI makes sense as a tool for "code generation". For the end-user experience, just using a C# API to generate C# code would not make sense and would feel a bit weird for the developer experience used to .NET approaches.
+- A CLI is aligned to modern cross-platform frameworks experience such as .NET Core, Python, Node, etc. 
+
+## Positioning the CLI and AutoML in ML.NET
+
+![alt text](Images/MLNET-AutoML-Positioning.png "ML.NET and AutoML")
+
+As mentioned at the begining of the spec doc, the CLI will be branded as the ML.NET CLI since this CLI will also have additional features where AutoML is not needed.
+
+The .NET AutoML API will be used for the most important foundational features that will be part of the ML.NET CLI, though, as shown in the following diagram:
+
+![alt text](Images/MLNET-AutoML-Working-Diagram.png "AutoML working inside the ML.NET CLI")
+
+
+## Design criteria
+
+- The ML.NET CLI will create **four main types of results** related to AutoML:
+    - Best trained models (Serialized model files, such as .ZIP files).
+    - C# ML.NET sample code consuming/scoring the models.
+    - C# ML.NET pipeline implementation code for each possible "best model" configuration.
+    - Result report with ML analysis of the models (*Mostly since v1.0*)
+
+- The ML.NET CLI automation will be able to run locally, on any development environment PC (Windows, Mac or Linux).
+- Future versions will be able to run AutoML compute and other compute processes (such as a regular model training) in Azure.
+- The ML.NET CLI  will consume the AutoML API (Microsoft.ML.Auto NuGet package) which will only consume public surface of ML.NET.
+- The CLI proposed here will not provide for “continue sweeping” after sweeping has ended.
```

Comment:
> The CLI proposed here will not provide for “continue sweeping” after sweeping has ended. [](start = 1, length = 89)

this should go under future features. This is a valuable functionality, but not needed for V1